ArduPlane/radio.cpp

```diff
@@ -353,20 +355,27 @@ bool Plane::trim_radio()
 }
 
 /*
-  return true if throttle level is below throttle failsafe threshold
-  or RC input is invalid
+  check if throttle input is within the valid range
  */
-bool Plane::rc_failsafe_active(void) const
+bool Plane::rc_throttle_in_range(void) const
 {
     if (!g.throttle_fs_enabled) {
-        return false;
-    }
-    if (millis() - failsafe.last_valid_rc_ms > 1000) {
-        // we haven't had a valid RC frame for 1 seconds
         return true;
     }
     if (channel_throttle->get_reverse()) {
-        return channel_throttle->get_radio_in() >= g.throttle_fs_value;
+        return channel_throttle->get_radio_in() < g.throttle_fs_value;
+    }
+    return channel_throttle->get_radio_in() > g.throttle_fs_value;
+}
+
+/*
+  return true if throttle level is below throttle failsafe threshold
+  or RC input is invalid
+ */
+bool Plane::rc_failsafe_active(void) const
+{
+    if (!rc_throttle_in_range()) {
+        return true;
```

Comment:
There is a slight change here: before, when throttle failsafe was disabled, it wouldn't check the time difference and now it does. It does work though because if throttle failsafe is disabled then last time is always updated.

It would enter into radio failsafe if the `read_radio` method wasn't called for more than 1 second though, but I think that's probably fine.

Reply:
I've checked, and it is OK, so I'll merge this